nixos/modules/services/networking/syncthing.nix

```diff
@@ -88,30 +117,53 @@ in
         config.ids.gids.syncthing;
     };
 
-    systemd.services.syncthing =
-      {
-        description = "Syncthing service";
-        after    = [ "network.target" ];
+    environment.systemPackages = [ cfg.package ] ++ lib.optional cfg.useInotify pkgs.syncthing-inotify;
+
+    systemd.services = mkIf cfg.systemService {
+      syncthing = header // {
+        after = [ "network.target" ];
         wantedBy = [ "multi-user.target" ];
-        environment = {
-          STNORESTART = "yes";  # do not self-restart
-          STNOUPGRADE = "yes";
-          inherit (cfg) all_proxy;
-        } // config.networking.proxy.envVars;
+        serviceConfig = service // {
+          User = cfg.user;
+          Group = optionalString (cfg.user == defaultUser) defaultUser;
+          PermissionsStartOnly = true;
+          ExecStart = "${pkgs.syncthing}/bin/syncthing -no-browser -home=${cfg.dataDir}";
+        };
+      };
 
+      syncthing-inotify = mkIf cfg.useInotify header // {
+        description = "Monitor Syncthing using inotify";
+        after = [ "syncthing.service" ];
+        requires = [ "syncthing.service" ];
+        wantedBy = [ "multi-user.target" ];
         serviceConfig = {
-          User  = cfg.user;
+          User = cfg.user;
           Group = optionalString (cfg.user == defaultUser) defaultUser;
           PermissionsStartOnly = true;
+          ExecStart = "${pkgs.syncthing-inotify}/bin/syncthing-inotify -logflags=0";
```

Comment:
It seems to me like the hard-coded use of `pkgs.syncthing-inotify` here completely breaks the abstraction of `cfg.package`. If I'd define `package` to be a variant of `syncthing` (with inotify support enabled), then I'd expect `syncthing-inotify.service` to use *that* package, not `pkgs.syncthing-inotify`.